_posts/2023-06-06-Client-cert-auth-without-roles.adoc

```diff
@@ -0,0 +1,252 @@
+`elytron-examples/client-cert-auth-without-roles/simple-webapp` folder.
+
+Please navigate to the elytron server home directory and enter the following command.
+
+```
+
+$SERVER_HOME/bin/jboss-cli.sh --connect 
+
+```
+
+This will connect you to the server, after which you may proceed to configuring the server to establish the connection. 
+
+First we need to create two keystores inside the elytron subsystem. This will tell elytron where to go inder to find the 
+keystores. The command for this is: 
+
+```
+/subsystem=elytron/key-store=twoWayKS:add(path=/path/to/server.keystore, relative-to=jboss.server.config.dir, credential-reference={clear-text=keystorepass}, type=JKS)
+/subsystem=elytron/key-store=twoWayTS:add(path=/path/to/server.truststore,relative-to=jboss.server.config.dir, credential-reference={clear-text=truststorepass}, type=JKS)
+```
+
+The `relative-to` option specifies that the path that you have entered is relative to the `/standalone/configuration` folder 
+inside your wildfly repository. 
+
+Next, we need to create a key-manager and a trust manager to point to `twoWayKS` and `twoWayTS`, the previously 
+created keystores using the following commands: 
+```
+/subsystem=elytron/key-manager=twoWayKM:add(key-store=twoWayKS,credential-reference={clear-text=keystorepass})
+
+/subsystem=elytron/trust-manager=twoWayTM:add(key-store=twoWayTS)
+```
+
+We also need to have an identity realm, which would help up map our user to one that exists inside the file system. 
+For this step, let us create a filesystem realm called `idsRealm`. We will enter into this realm our identity with 
+name `client-cert-user` which we will map the certificate into. 
+
+```
+/subsystem=elytron/filesystem-realm=idsRealm:add(path=ids, relative-to=jboss.server.config.dir)
+/subsystem=elytron/filesystem-realm=idsRealm:add-identity(identity=client-cert-user)
+```
+
+Next, we need to configure our server-ssl-context. This is a collection of ciphers, TLS settings and certificates, like the 
+ones we will be using to authenticate our server and the client. Since the `applicationSSC` server-SSL-context already exists, 
+we will not be adding one. Instead, we will be editing its properties using the `write-attribute` command: 
+
+```
+/subsystem=elytron/server-ssl-context=applicationSSC:write-attribute(name=key-manager, value=twoWayKM)
+/subsystem=elytron/server-ssl-context=applicationSSC:write-attribute(name=trust-manager, value=twoWayTM)
+/subsystem=elytron/server-ssl-context=applicationSSC:write-attribute(name=need-client-auth, value=true)
+```
+
+Now, when we are establishing the connection, elytron will know where to go to look for the keystore and the truststore. 
+Setting `want-client-auth` to true indicates that a client needs to be authenticated in order to establish a connection. 
+
+Next, we will need to configure a decoder that will be able to decode the CA Certification. We will be using 
```

Comment:
This could be reworded to something like:

Next, we will configure a principal transformer that will map an X.509 certificate to a generic user name. To do this, we will make use of a `chained-principal-transformer` that consists of an `x500-attribute-principal-decoder` and a `regex-principal-transformer`.

The `x500-attribute-principal-decoder` is used to obtain the first segment from the CN value from an X.509 certificate.

The `regex-principal-transformer` checks if the CN value matches a specific pattern and if so maps the principal to "client-cert-user".